test/langtools/tools/lib/toolbox/ToolBox.java

```diff
@@ -180,9 +180,41 @@ public void checkEqual(List<String> l1, List<String> l2) throws Error {
      * @return the strings matching the regular expression
      */
     public List<String> grep(Pattern pattern, List<String> lines) {
-        return lines.stream()
-                .filter(s -> pattern.matcher(s).find())
-                .collect(Collectors.toList());
+        return grep(pattern, lines, false);
+    }
+
+    /**
+     * Filters a list of strings according to the given regular expression.
+     * @param regex the regular expression
+     * @param lines the strings to be filtered
+     * @param invert identify positive or negative filtering
+     *               true: negative filtering, return the unmatched strings
+     *               false: positive filtering, return the matched strings
+     * @return the strings matching(or not matching) the regular expression
+     */
+    public List<String> grep(String regex, List<String> lines, boolean invert) {
+        return grep(Pattern.compile(regex), lines, invert);
+    }
+
+    /**
+     * Filters a list of strings according to the given regular expression.
+     * @param pattern the regular expression
+     * @param lines the strings to be filtered
+     * @param invert identify positive or negative filtering
+     *               true: negative filtering, return the unmatched strings
+     *               false: positive filtering, return the matched strings
+     * @return the strings matching(or not matching) the regular expression
+     */
+    public List<String> grep(Pattern pattern, List<String> lines, boolean invert) {
```

Comment:
what about?

```
    public List<String> grep(Pattern pattern, List<String> lines, boolean invert) {
        return lines.stream()
                .filter(s -> invert ? !pattern.matcher(s).find() : pattern.matcher(s).find())
                .collect(Collectors.toList());
    }
```

Reply:
Putting the `invert` into method `filter` may cause a little performance influence. Is it similar to `loop`? 
Because this is the test code, I adopt your suggestion.